src/base/Repr.hh

```diff
@@ -0,0 +1,445 @@
+ * into test code.  The default tries to use whatever ostream operator is
+ * available.
+ * Other overrides are provided for collections, characters, and more?
+ */
+template<class T>
+struct ReprTraits
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "UNKNOWN", name);
+    }
+    static void init(std::ostream&) {}
+    static void print_value(std::ostream& os, const T& value) { os << value; }
+};
+
+template<>
+struct ReprTraits<float>
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "float", name);
+    }
+    static void init(std::ostream& os) { os.precision(7); }
+    static void print_value(std::ostream& os, float value)
+    {
+        os << value << 'f';
+    }
+};
+
+template<>
+struct ReprTraits<double>
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "double", name);
+    }
+    static void init(std::ostream& os) { os.precision(14); }
+    static void print_value(std::ostream& os, double value) { os << value; }
+};
+
+template<>
+struct ReprTraits<char>
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "char", name);
+    }
+    static void init(std::ostream&) {}
+    static void print_value(std::ostream& os, char value)
+    {
+        os << '\'';
+        detail::repr_char(os, value);
+        os << '\'';
+    }
+};
+
+template<>
+struct ReprTraits<unsigned char>
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "unsigned char", name);
+    }
+    static void init(std::ostream& os) { os << std::setfill('0') << std::hex; }
+    static void print_value(std::ostream& os, unsigned char value)
+    {
+        os << "'\\x" << detail::char_to_hex_string(value) << '\'';
+    }
+};
+
+template<>
+struct ReprTraits<bool>
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "bool", name);
+    }
+    static void init(std::ostream& os) { os << std::boolalpha; }
+    static void print_value(std::ostream& os, bool value) { os << value; }
+};
+
+template<>
+struct ReprTraits<int>
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "int", name);
+    }
+    static void init(std::ostream&) {}
+    static void print_value(std::ostream& os, int value) { os << value; }
+};
+
+template<>
+struct ReprTraits<unsigned int>
+{
+    static void print_type(std::ostream& os, const char* name = nullptr)
+    {
+        detail::print_simple_type(os, "unsigned int", name);
+    }
+    static void init(std::ostream&) {}
+    static void print_value(std::ostream& os, unsigned int value)
+    {
+        os << value << 'u';
+    }
+};
+
+#ifndef _WIN32
```

Comment:
Why is this a conditional on 'windows'?  Is there a more 'generic' case where those are not needed?

Reply:
This is to fix multiple definition errors on windows where `long` is the same as `int` (LLP64). There's probably a better way of doing this, as I don't know if the error I got when I put this in years ago was because of the standard library, the compiler, the windows version, or the windows platform itself. But it works at least? Though since Celeritas doesn't have any windows users we could probably delete it.

Reply:
I agree that we should remove the `#ifndef _WIN32` the issue is plausibly fixed and/or gone anyway with newer version of the Microsoft compiler and/or the fact that if/when we port to windows if would be 64bits rather than 32bits.

Reply:
Well, this is actually an issue with 64-bit windows: the _WIN32 macro is defined on all windows machines: it's very confusing. But since we're not using this code on windows yet (and our downstream windows code isn't using Celeritas yet) I'll remove the ifdef and see if it still breaks eventually, and if/when that happens we can seek a less ugly solution.